rotkehlchen/tests/api/test_caching.py

```diff
@@ -0,0 +1,181 @@
+    with open(f'{icons_dir}/ETH_small.png', 'wb') as f:
+        f.write(b'')
+
+    with open(f'{icons_dir}/BTC_small.png', 'wb') as f:
+        f.write(b'')
+
+    with open(f'{icons_dir}/AVAX_small.png', 'wb') as f:
+        f.write(b'')
+
+    # also add an avatar, to make sure it's not deleted when icons are
+    with open(f'{avatars_dir}/me.eth.png', 'wb') as f:
+        f.write(b'')
+
+    assert len([i for i in icons_dir.iterdir() if i.is_file()]) == 3
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'clearcacheresource',
+            cache_type=CacheInvalidationType.ICONS,
+        ), json={
+            'entries': ['ETH'],
+        },
+    )
+    assert_proper_response(response)
+    assert len([i for i in icons_dir.iterdir() if i.is_file()]) == 2
+    assert len([i for i in avatars_dir.iterdir() if i.is_file()]) == 1
+
+    # delete icons cache completely
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'clearcacheresource',
+            cache_type=CacheInvalidationType.ICONS,
+        ),
+    )
+    assert_proper_response(response)
+    assert len([i for i in icons_dir.iterdir() if i.is_file()]) == 0
+    assert len([i for i in avatars_dir.iterdir() if i.is_file()]) == 1
+
+    # populate avatars dir to test the cache deletion
+    with open(f'{avatars_dir}/ava.eth.png', 'wb') as f:
+        f.write(b'')
+
+    with open(f'{avatars_dir}/prettyirrelevant.eth.png', 'wb') as f:
+        f.write(b'')
+
+    with open(f'{avatars_dir}/nebolax.eth.png', 'wb') as f:
+        f.write(b'')
+
+    assert len([i for i in avatars_dir.iterdir() if i.is_file()]) == 4
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'clearcacheresource',
+            cache_type=CacheInvalidationType.AVATARS,
+        ), json={
+            'entries': ['ava.eth'],
+        },
+    )
+    assert_proper_response(response)
+    assert len([i for i in avatars_dir.iterdir() if i.is_file()]) == 3
+
+    # delete avatars cache completely
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'clearcacheresource',
+            cache_type=CacheInvalidationType.AVATARS,
+        ),
+    )
+    assert_proper_response(response)
+    assert len([i for i in avatars_dir.iterdir() if i.is_file()]) == 0
+
+    # now try fetching the avatar and see it works
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server,
+            'ensavatarsresource',
+            ens_name='nebolax.eth',
+        ),
+    )
+    assert response.status_code == 200
+    assert response.headers['Content-Type'] == 'image/png'
+
+
+def test_general_cache_invalidation(rotkehlchen_api_server: 'APIServer'):
+    """Tests that invalidating the general cache works as expected"""
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
+    old_pool = CurvePoolData(
```

Comment:
this is only testing curve. I would patch the 3 functions that you call and assert that they got called. There is `times_called` or something similar once you magicmock. Each of those functions should be already tested. Otherwise you are double testing

Reply:
Good suggestion, thanks